<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Raphael Eisenhofer" />


<title>Supplemental analyses for reviewer #1</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.20/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2020_Koala_Faeces_R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/EisenRa/2020_Koala_Faeces_R">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Supplemental analyses for reviewer #1</h1>
<h4 class="author">Raphael Eisenhofer</h4>
<h4 class="date">Oct 11, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>2020_Koala_Faeces_R/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201209code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20201209)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201209code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201209)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomEisenRa2020KoalaFaecesRtreed7fbe1402a30308a1ce60401bc3259c9551712a6targetblankd7fbe14a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/EisenRa/2020_Koala_Faeces_R/tree/d7fbe1402a30308a1ce60401bc3259c9551712a6" target="_blank">d7fbe14</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomEisenRa2020KoalaFaecesRtreed7fbe1402a30308a1ce60401bc3259c9551712a6targetblankd7fbe14a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/EisenRa/2020_Koala_Faeces_R/tree/d7fbe1402a30308a1ce60401bc3259c9551712a6" target="_blank">d7fbe14</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .pcoa_uwUniFrac_all_Koala_PC1-2.png
    Untracked:  AlphaDiv.png
    Untracked:  R_analyses.Rmd
    Untracked:  Residuals_histo_evenness.png
    Untracked:  Residuals_histo_richness.png
    Untracked:  Rplot.png
    Untracked:  alpha_diversity_results.csv
    Untracked:  analysis/Supplemental_analysis_for_reviewer1.Rmd
    Untracked:  evenness_location.html
    Untracked:  output/ASVresults.csv
    Untracked:  output/AbundancePrevalence_Per_Phylum.png
    Untracked:  output/FIGURES/
    Untracked:  output/Figure1A.svg
    Untracked:  output/Figure1B.svg
    Untracked:  output/Figure2.pdf
    Untracked:  output/Figure2.png
    Untracked:  output/Figure2.svg
    Untracked:  output/Figure3A.svg
    Untracked:  output/Figure3B.svg
    Untracked:  output/Figure4.pdf
    Untracked:  output/Figure4.png
    Untracked:  output/Figure4.svg
    Untracked:  output/Figure_3.pdf
    Untracked:  output/Figure_3.png
    Untracked:  output/Figure_4.pdf
    Untracked:  output/Figure_4.png
    Untracked:  output/Map_Figure.png
    Untracked:  output/Map_Figure.pptx
    Untracked:  output/R_Aus_Map.svg
    Untracked:  output/SIX.png
    Untracked:  output/SI_Figure1.png
    Untracked:  output/SI_Figure1.svg
    Untracked:  output/SI_Figure2A.svg
    Untracked:  output/SI_Figure2B.svg
    Untracked:  output/SI_Figure_3.png
    Untracked:  output/SI_figure1.pdf
    Untracked:  output/SI_figure2.pdf
    Untracked:  output/SI_figure3.pdf.2022-07-01_21-13-37.log
    Untracked:  output/SI_figure3.pdf.2022-07-01_21-14-22.log
    Untracked:  output/SI_figure3.png
    Untracked:  output/SI_figure3.png.2022-07-01_21-16-04.log
    Untracked:  output/SI_figure3.png.2022-09-22_13-00-07.log
    Untracked:  output/SI_figure3.png.2022-10-10_10-11-42.log
    Untracked:  output/SI_figure3.png.2022-10-10_10-13-08.log
    Untracked:  output/SI_figure3.png.2022-10-10_10-18-18.log
    Untracked:  output/SI_figure3.png.2022-10-10_10-28-56.log
    Untracked:  output/SI_figure3.png.2022-10-10_10-31-13.log
    Untracked:  output/SI_figure3.png.2022-10-11_10-08-53.log
    Untracked:  output/SI_figure3.png.2022-10-11_10-13-17.log
    Untracked:  output/SI_figure3.png.2022-10-11_10-16-33.log
    Untracked:  output/SI_figure3.svg
    Untracked:  output/SI_figure3.svg.2022-07-01_21-14-50.log
    Untracked:  output/SI_figure3.tiff.2022-07-01_21-15-18.log
    Untracked:  output/SI_figure_4.pdf
    Untracked:  output/SI_koala_diversity.png
    Untracked:  output/SI_ppn_diversity.png
    Untracked:  output/SIkoala_diversity.png
    Untracked:  output/Venn.png
    Untracked:  output/Venn.png.2021-06-25_12-02-28.log
    Untracked:  output/Venn.tiff
    Untracked:  output/Venn.tiff.2021-06-23_12-12-29.log
    Untracked:  output/Venn.tiff.2021-06-23_12-13-44.log
    Untracked:  output/Venn.tiff.2021-06-23_12-14-18.log
    Untracked:  output/Venn.tiff.2021-06-25_11-59-24.log
    Untracked:  output/Venn.tiff.2021-06-25_12-02-02.log
    Untracked:  output/Venn.tiff.2022-02-02_13-43-38.log
    Untracked:  output/Venn.tiff.2022-02-02_14-23-01.log
    Untracked:  output/Venn.tiff.2022-04-18_12-38-01.log
    Untracked:  output/Venn.tiff.2022-04-18_12-45-23.log
    Untracked:  richness_location.html
    Untracked:  test.txt

Unstaged changes:
    Modified:   Misc_code.R
    Modified:   analysis/QIIME2/Koala_Metadata.tsv
    Modified:   analysis/R_analyses.Rmd
    Modified:   output/Family_Taxa_Plot.png
    Deleted:    output/pcoa_uwUniFrac_MtLagoon_Koala_PC1-4.png
    Modified:   output/pcoa_uwUniFrac_all_Population_PC1-2.png
    Modified:   output/pcoa_uwUniFrac_all_Population_PC1-3.png
    Deleted:    output/pcoa_wUniFrac_MtLagoon_Koala_PC1-4.png
    Modified:   output/pcoa_wUniFrac_all_Population_PC1-2.png
    Modified:   output/pcoa_wUniFrac_all_Population_PC1-3.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages-import-data" class="section level2">
<h2>Load packages, import data</h2>
<pre class="r"><code>library(phyloseq)
library(qiime2R)
library(ggplot2)
library(scales)
library(microbiome)
library(tidyverse)</code></pre>
<pre><code>Warning: package &#39;tidyverse&#39; was built under R version 4.1.3</code></pre>
<pre class="r"><code>library(ggmap)
library(ggrepel)
library(ggpubr)
library(VennDiagram)
library(osmdata)


#Import data into phyloseq object
ps &lt;- qza_to_phyloseq(
  features = &quot;analysis/QIIME2/Koala-table.qza&quot;,
  tree = &quot;analysis/QIIME2/Koala-sepp-tree.qza&quot;,
  taxonomy = &quot;analysis/QIIME2/Koala-SILVA-138.qza&quot;,
  metadata = &quot;analysis/QIIME2/Koala_Metadata.tsv&quot;
)</code></pre>
</div>
<div
id="filter-out-asvs-lower-than-0.25-relative-abundance-for-reviewer-1"
class="section level1">
<h1>Filter out ASVs lower than 0.25% relative abundance (for reviewer
#1)</h1>
<pre class="r"><code>#Rarefy data
ps.rar &lt;- rarefy_even_depth(ps, sample.size = 36622, rngseed = 1337)</code></pre>
<pre><code>`set.seed(1337)` was used to initialize repeatable random subsampling.</code></pre>
<pre><code>Please record this for your records so others can reproduce.</code></pre>
<pre><code>Try `set.seed(1337); .Random.seed` for the full vector</code></pre>
<pre><code>...</code></pre>
<pre><code>11OTUs were removed because they are no longer 
present in any sample after random subsampling</code></pre>
<pre><code>...</code></pre>
<pre class="r"><code>#Apply 0.25% relative abundance filter (entire OTU table)
count_cutoff &lt;- 36622 * 0.0025
ps.rar.tablefilter &lt;- filter_taxa(ps.rar, function(x) sum(x &gt; count_cutoff) &gt; (0.0125*length(x)), TRUE)

### 732 ASVs were pruned with this threshold, and these ASVs account for &lt; 2% of the total reads in the dataset</code></pre>
</div>
<div id="alpha-diversity-through-time-plots-figure-2-si-figures-x-x"
class="section level1">
<h1>Alpha diversity through time plots (Figure 2; SI Figures X &amp;
X)</h1>
<pre class="r"><code>#Calculate alpha diversities
ps.rar.tablefilter.div &lt;- alpha(ps.rar.tablefilter, index = &quot;all&quot;)</code></pre>
<pre><code>Observed richness</code></pre>
<pre><code>Other forms of richness</code></pre>
<pre><code>Diversity</code></pre>
<pre><code>Evenness</code></pre>
<pre><code>Dominance</code></pre>
<pre><code>Rarity</code></pre>
<pre class="r"><code>#Append metadata to alpha diversity values
ps.rar.tablefilter.meta &lt;- meta(ps.rar.tablefilter)
ps.rar.tablefilter.meta$name &lt;- rownames(ps.rar.tablefilter.meta)
ps.rar.tablefilter.div$name &lt;- rownames(ps.rar.tablefilter.div)
ps.rar.tablefilter.div.df &lt;- merge(ps.rar.tablefilter.div, ps.rar.tablefilter.meta, by = &quot;name&quot;)


#Set ggplot theme
theme_RE &lt;- theme(axis.text.x = element_text(size=20), 
                  axis.text.y = element_text(size=20),
                  axis.title.x = element_text(size=24, face=&quot;bold&quot;),
                  axis.title.y = element_text(size=24, face=&quot;bold&quot;),
                  axis.line = element_line(colour = &quot;black&quot;),
                  #Background panel
                  panel.background = element_rect(fill = &quot;White&quot;),
                  panel.grid.major = element_line(colour = &quot;white&quot;), 
                  panel.grid.minor = element_line(colour = &quot;white&quot;),
                  #Legend
                  legend.title = element_blank(),
                  legend.text = element_text(size=16),
                  legend.key = element_rect(fill = &quot;white&quot;, color = NA),
                  legend.position = c(0.5, 0.18),
                  legend.direction = &quot;horizontal&quot;,
                  legend.key.size = unit(3.5, &quot;line&quot;))

#Set colours
cbp2_cleland &lt;- c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;, 
                  &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;)

cbp2_mtlagoon &lt;- c(&quot;#88CCEE&quot;, &quot;#CC6677&quot;, &quot;#DDCC77&quot;, &quot;#117733&quot;, &quot;#998ec3&quot;)


#Create each pane of figure 2
plot.evenness.cleland &lt;- ps.rar.tablefilter.div.df %&gt;%
  filter(Location_sampled == &quot;Cleland&quot;) %&gt;%
  ggplot(aes(x=Collection_number, y=evenness_pielou, color=Koala)) +
  geom_line(size = 3) +
  geom_point(size = 5) +
  scale_x_continuous(breaks = seq(1, 8, 1), labels = c(
    &quot;1&quot; = &quot;4&quot;, &quot;2&quot; = &quot;8&quot;, &quot;3&quot; = &quot;12&quot;, &quot;4&quot; = &quot;16&quot;, 
    &quot;5&quot; = &quot;20&quot;, &quot;6&quot; = &quot;24&quot;, &quot;7&quot; = &quot;28&quot;, &quot;8&quot; = &quot;32&quot;
  )) +
  scale_y_continuous(limits = c(0, 1), breaks = seq(0, 1, 0.2)) +
  xlab(&quot; &quot;) +
  ylab(&quot;pielou evenness&quot;) +
  scale_color_manual(values = cbp2_cleland) +
  theme_RE


plot.evenness.mtlagoon &lt;- ps.rar.tablefilter.div.df %&gt;%
  filter(Location_sampled != &quot;Cleland&quot;) %&gt;%
  ggplot(aes(x=Collection_number, y=evenness_pielou, color=Koala)) +
  geom_line(size = 3) +
  geom_point(size = 5) +
  scale_x_continuous(breaks = seq(1, 8, 1)) +
  scale_y_continuous(limits = c(0, 1), breaks = seq(0, 1, 0.2)) +
  xlab(&quot; &quot;) +
  ylab(&quot; &quot;) +
  scale_color_manual(values = cbp2_mtlagoon) +
  theme_RE


plot.richness.cleland &lt;- ps.rar.tablefilter.div.df %&gt;%
  filter(Location_sampled == &quot;Cleland&quot;) %&gt;%
  ggplot(aes(x=Collection_number, y=observed, color=Koala)) +
  geom_line(size = 3) +
  geom_point(size = 5) +
  scale_x_continuous(breaks = seq(1, 8, 1), labels = c(
    &quot;1&quot; = &quot;4&quot;, &quot;2&quot; = &quot;8&quot;, &quot;3&quot; = &quot;12&quot;, &quot;4&quot; = &quot;16&quot;, 
    &quot;5&quot; = &quot;20&quot;, &quot;6&quot; = &quot;24&quot;, &quot;7&quot; = &quot;28&quot;, &quot;8&quot; = &quot;32&quot;
  )) +
  scale_y_continuous(limits = c(1, 100), breaks = seq(0, 100, 25)) +
  xlab(&quot;Day&quot;) +
  ylab(&quot;ASV richness&quot;) +
  scale_color_manual(values = cbp2_cleland) +
  theme_RE +
  theme(legend.position=&#39;none&#39;)


plot.richness.mtlagoon &lt;- ps.rar.tablefilter.div.df %&gt;%
  filter(Location_sampled != &quot;Cleland&quot;) %&gt;%
  ggplot(aes(x=Collection_number, y=observed, color=Koala)) +
  geom_line(size = 3) +
  geom_point(size = 5) +
  scale_x_continuous(breaks = seq(1, 8, 1)) +
  scale_y_continuous(limits = c(1, 100), breaks = seq(0, 100, 25)) +
  xlab(&quot;Month&quot;) +
  ylab(&quot; &quot;) +
  scale_color_manual(values = cbp2_mtlagoon) +
  theme_RE +
  theme(legend.position=&#39;none&#39;) 

#Merge panes into figure 2
ggarrange(plot.evenness.cleland, plot.evenness.mtlagoon,
          plot.richness.cleland, plot.richness.mtlagoon,
          nrow = 2, ncol = 2, common.legend = FALSE, labels = c(&quot;A)&quot;, &quot;B)&quot;, &quot;C)&quot;, &quot;D)&quot;), 
          font.label = list(size=18, face=&quot;bold&quot;, color=&quot;black&quot;))</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/Supplemental_analysis_for_reviewer1.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(filename = &quot;output/Figure2.pdf&quot;, 
#        width = 16, height = 14, dpi = 300)

##########################################################################################

# Figures SX and SX

#Order samples
ps.rar.tablefilter.div.df$Koala &lt;- factor(ps.rar.tablefilter.div.df$Koala, 
                                          levels = c(&#39;Blaze&#39;, &#39;Byron&#39;, &#39;Claudette&#39;, &#39;George&#39;, 
                                                    &#39;Leo&#39;, &#39;Lola&#39;, &#39;Theo&#39;,
                                                    &#39;Cin&#39;, &#39;Phasco&#39;, &#39;Reus&#39;, &#39;Tos&#39;, &#39;Uno&#39;))
ps.rar.tablefilter.div.df$Koala &lt;- factor(ps.rar.tablefilter.div.df$Koala, 
                                          levels = c(&#39;Blaze&#39;, &#39;Byron&#39;, &#39;Claudette&#39;, &#39;George&#39;, 
                                                    &#39;Leo&#39;, &#39;Lola&#39;, &#39;Theo&#39;,
                                                    &#39;Cin&#39;, &#39;Phasco&#39;, &#39;Reus&#39;, &#39;Tos&#39;, &#39;Uno&#39;))


#Colour theme all
cbp2_all &lt;- c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;, &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;,
              &quot;#88CCEE&quot;, &quot;#CC6677&quot;, &quot;#DDCC77&quot;, &quot;#117733&quot;, &quot;#998ec3&quot;)


koala.richness &lt;- ggplot(ps.rar.tablefilter.div.df, aes(x=Koala, y=observed))
koala.evenness &lt;- ggplot(ps.rar.tablefilter.div.df, aes(x=Koala, y=evenness_pielou))

#Set theme
theme_RE2 &lt;- theme(axis.text.x = element_text(face=&quot;bold&quot;, size=14, angle = 45, vjust = 0.7), 
                   axis.text.y = element_text(face=&quot;bold&quot;, size=20),
                   axis.title.x = element_text(size=20, face=&quot;bold&quot;),
                   axis.title.y = element_text(size=20, face=&quot;bold&quot;),
                   axis.line = element_line(colour = &quot;black&quot;),
                   panel.background = element_rect(fill = &quot;White&quot;),
                   strip.text = element_text(colour = &quot;black&quot;, face=&quot;bold&quot;, size = 14),
                   #Legend
                   legend.position = &quot;none&quot;)

#Create panes
gg.koala.richness = koala.richness + 
  facet_wrap(~Location_sampled, scales = &quot;free_x&quot;) +
  geom_boxplot(size=1, outlier.shape=8, outlier.size=3, outlier.stroke=1, aes(colour=Koala)) +
  geom_jitter(position=position_dodge2(0.3), size=3.5, aes(colour=Koala)) +
  scale_colour_manual(values=cbp2_all) +
  theme_RE2 +
  labs(x = &quot;&quot;) +
  labs(y = &quot;ASV Richness&quot;)

gg.koala.evenness = koala.evenness + 
  facet_wrap(~Location_sampled, scales = &quot;free_x&quot;) +
  geom_boxplot(size=1, outlier.shape=8, outlier.size=3, outlier.stroke=1, aes(colour=Koala)) +
  geom_jitter(position=position_dodge2(0.3), size=3.5, aes(colour=Koala)) +
  scale_colour_manual(values=cbp2_all) +
  theme_RE2 +
  labs(x = &quot;&quot;) +
  labs(y = &quot;Pielou evenness&quot;)

#Merge panes into SI figure X
ggarrange(gg.koala.evenness, gg.koala.richness,
          nrow = 2, ncol = 1, common.legend = FALSE, labels = c(&quot;A)&quot;, &quot;B)&quot;), 
          font.label = list(size=18, face=&quot;bold&quot;, color=&quot;black&quot;))</code></pre>
<p><img src="figure/Supplemental_analysis_for_reviewer1.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(filename = &quot;output/SI_figure1.pdf&quot;, 
#        width = 16, height = 10, dpi = 300)

##########################################################################################

# Cleland vs. Mountain Lagoon
ppn.richness &lt;- ggplot(ps.rar.tablefilter.div.df, aes(x=Location_sampled, y=observed))
ppn.evenness &lt;- ggplot(ps.rar.tablefilter.div.df, aes(x=Location_sampled, y=evenness_pielou))

#Create panes
gg.ppn.richness = ppn.richness + 
  geom_boxplot(size=1, outlier.shape=8, outlier.size=3, outlier.stroke=1, aes(colour=Location_sampled)) +
  geom_jitter(position=position_dodge2(0.3), size=3.5, aes(colour=Location_sampled)) +
  scale_colour_manual(values=cbp2_all) +
  theme_RE2 +
  labs(x = &quot;&quot;) +
  labs(y = &quot;ASV Richness&quot;)

gg.ppn.evenness = ppn.evenness + 
  geom_boxplot(size=1, outlier.shape=8, outlier.size=3, outlier.stroke=1, aes(colour=Location_sampled)) +
  geom_jitter(position=position_dodge2(0.3), size=3.5, aes(colour=Location_sampled)) +
  scale_colour_manual(values=cbp2_all) +
  theme_RE2 +
  labs(x = &quot;&quot;) +
  labs(y = &quot;Pielou evenness&quot;)

#Merge panes into SI figure X
ggarrange(gg.ppn.evenness, gg.ppn.richness,
          nrow = 1, ncol = 2, common.legend = FALSE, labels = c(&quot;A)&quot;, &quot;B)&quot;), 
          font.label = list(size=18, face=&quot;bold&quot;, color=&quot;black&quot;))</code></pre>
<p><img src="figure/Supplemental_analysis_for_reviewer1.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(filename = &quot;output/SI_figure2.pdf&quot;, 
#        width = 16, height = 10, dpi = 300)</code></pre>
</div>
<div id="alpha-diversity-stats" class="section level1">
<h1>Alpha diversity stats</h1>
<pre class="r"><code>library(lme4)</code></pre>
<pre><code>Loading required package: Matrix</code></pre>
<pre><code>
Attaching package: &#39;Matrix&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:tidyr&#39;:

    expand, pack, unpack</code></pre>
<pre class="r"><code>library(lmerTest)</code></pre>
<pre><code>
Attaching package: &#39;lmerTest&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:lme4&#39;:

    lmer</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    step</code></pre>
<pre class="r"><code>library(stargazer)</code></pre>
<pre><code>
Please cite as: </code></pre>
<pre><code> Hlavac, Marek (2018). stargazer: Well-Formatted Regression and Summary Statistics Tables.</code></pre>
<pre><code> R package version 5.2.2. https://CRAN.R-project.org/package=stargazer </code></pre>
<pre class="r"><code>##Linear model of diversity ~ koala
koala.richness &lt;- lm(rank(observed) ~ Koala, data = ps.rar.tablefilter.div.df)
koala.evenness &lt;- lm(rank(evenness_pielou) ~ Koala, data = ps.rar.tablefilter.div.df)

#ANOVA
anova(koala.richness)</code></pre>
<pre><code>Analysis of Variance Table

Response: rank(observed)
          Df Sum Sq Mean Sq F value    Pr(&gt;F)    
Koala     11  35387  3217.0   30.31 &lt; 2.2e-16 ***
Residuals 68   7217   106.1                      
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>anova(koala.evenness)</code></pre>
<pre><code>Analysis of Variance Table

Response: rank(evenness_pielou)
          Df Sum Sq Mean Sq F value    Pr(&gt;F)    
Koala     11  23068 2097.06  7.2784 4.797e-08 ***
Residuals 68  19592  288.12                      
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>#Do the residuals conform to normality?
hist(residuals(koala.richness))</code></pre>
<p><img src="figure/Supplemental_analysis_for_reviewer1.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist(residuals(koala.evenness))</code></pre>
<p><img src="figure/Supplemental_analysis_for_reviewer1.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>shapiro.test(residuals(koala.richness))</code></pre>
<pre><code>
    Shapiro-Wilk normality test

data:  residuals(koala.richness)
W = 0.98409, p-value = 0.4235</code></pre>
<pre class="r"><code>shapiro.test(residuals(koala.evenness))</code></pre>
<pre><code>
    Shapiro-Wilk normality test

data:  residuals(koala.evenness)
W = 0.9748, p-value = 0.1149</code></pre>
<pre class="r"><code>##Linear model of diversity ~ location sampled (Cleland vs. Mountain Lagoon)
#Using koala as a random effect to account for individual effects
location.richness &lt;- lmer(rank(observed) ~ Location_sampled + (1|Koala), data = ps.rar.tablefilter.div.df)
location.evenness &lt;- lmer(rank(evenness_pielou) ~ Location_sampled + (1|Koala), data = ps.rar.tablefilter.div.df)

summary(location.richness)</code></pre>
<pre><code>Linear mixed model fit by REML. t-tests use Satterthwaite&#39;s method [
lmerModLmerTest]
Formula: rank(observed) ~ Location_sampled + (1 | Koala)
   Data: ps.rar.tablefilter.div.df

REML criterion at convergence: 613.7

Scaled residuals: 
     Min       1Q   Median       3Q      Max 
-2.34091 -0.54838 -0.02594  0.55563  2.64415 

Random effects:
 Groups   Name        Variance Std.Dev.
 Koala    (Intercept) 123.7    11.12   
 Residual             105.8    10.29   
Number of obs: 80, groups:  Koala, 12

Fixed effects:
                                Estimate Std. Error      df t value Pr(&gt;|t|)
(Intercept)                       52.098      4.422   9.854  11.781 3.98e-07
Location_sampledMountain_Lagoon  -38.513      6.992  10.671  -5.508 0.000205
                                   
(Intercept)                     ***
Location_sampledMountain_Lagoon ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Correlation of Fixed Effects:
            (Intr)
Lctn_smpM_L -0.632</code></pre>
<pre class="r"><code>summary(location.evenness)</code></pre>
<pre><code>Linear mixed model fit by REML. t-tests use Satterthwaite&#39;s method [
lmerModLmerTest]
Formula: rank(evenness_pielou) ~ Location_sampled + (1 | Koala)
   Data: ps.rar.tablefilter.div.df

REML criterion at convergence: 689.8

Scaled residuals: 
     Min       1Q   Median       3Q      Max 
-2.78990 -0.58178 -0.00444  0.64274  1.99960 

Random effects:
 Groups   Name        Variance Std.Dev.
 Koala    (Intercept) 267.9    16.37   
 Residual             288.2    16.98   
Number of obs: 80, groups:  Koala, 12

Fixed effects:
                                Estimate Std. Error     df t value Pr(&gt;|t|)    
(Intercept)                       36.929      6.589  9.364   5.605 0.000288 ***
Location_sampledMountain_Lagoon   13.353     10.465 10.318   1.276 0.229941    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Correlation of Fixed Effects:
            (Intr)
Lctn_smpM_L -0.630</code></pre>
<pre class="r"><code>#Tabluate
# stargazer(location.richness, type = &quot;html&quot;,
#           digits = 3,
#           star.cutoffs = c(0.05, 0.01, 0.001),
#           digit.separator = &quot;&quot;,
#           out = &quot;richness_location.html&quot;)

# stargazer(location.evenness, type = &quot;html&quot;,
#           digits = 3,
#           star.cutoffs = c(0.05, 0.01, 0.001),
#           digit.separator = &quot;&quot;,
#           out = &quot;evenness_location.html&quot;)

#Get mean+SD for Cleland vs. Mountain Lagoon
ps.rar.tablefilter.div.df %&gt;%
  group_by(Location_sampled) %&gt;%
  summarise(mean = mean(observed), sd = sd(observed))</code></pre>
<pre><code># A tibble: 2 x 3
  Location_sampled  mean    sd
  &lt;chr&gt;            &lt;dbl&gt; &lt;dbl&gt;
1 Cleland           70.6 13.6 
2 Mountain_Lagoon   47.0  5.68</code></pre>
</div>
<div id="pcoa-figures-figures-3-and-si-x" class="section level1">
<h1>PCoA figures (Figures 3, and SI X)</h1>
<pre class="r"><code>##Split table into cleland and mt lagoon
ps.rar.tablefilter.cl &lt;- subset_samples(ps.rar.tablefilter, Location_sampled == &quot;Cleland&quot;)
ps.rar.tablefilter.ml &lt;- subset_samples(ps.rar.tablefilter, Location_sampled == &quot;Mountain_Lagoon&quot;)

##calculate beta diversities
#Unweighted UniFrac
ord.unw.uni.cl &lt;- ordinate(ps.rar.tablefilter.cl, &quot;PCoA&quot;, &quot;unifrac&quot;, weighted=F)
ord.unw.uni.ml &lt;- ordinate(ps.rar.tablefilter.ml, &quot;PCoA&quot;, &quot;unifrac&quot;, weighted=F)

#Weighted UniFrac
ord.w.uni.cl &lt;- ordinate(ps.rar.tablefilter.cl, &quot;PCoA&quot;, &quot;unifrac&quot;, weighted=T)
ord.w.uni.ml &lt;- ordinate(ps.rar.tablefilter.ml, &quot;PCoA&quot;, &quot;unifrac&quot;, weighted=T)

#Axes 1/2
unwt.unifrac.1.2.cl &lt;- plot_ordination(ps.rar.tablefilter.cl, 
                                    ord.unw.uni.cl, color=&quot;Koala&quot;, 
                                    axes = c(1, 2))
unwt.unifrac.1.2.ml &lt;- plot_ordination(ps.rar.tablefilter.ml, 
                                    ord.unw.uni.ml, color=&quot;Koala&quot;, 
                                    axes = c(1, 2))

w.unifrac.1.2.cl &lt;- plot_ordination(ps.rar.tablefilter.cl, 
                                 ord.w.uni.cl, color=&quot;Koala&quot;, 
                                 axes = c(1, 2))
w.unifrac.1.2.ml &lt;- plot_ordination(ps.rar.tablefilter.ml, 
                                 ord.w.uni.ml, color=&quot;Koala&quot;, 
                                 axes = c(1, 2))
#Axes 1/3
unwt.unifrac.1.3.cl &lt;- plot_ordination(ps.rar.tablefilter.cl, 
                                    ord.unw.uni.cl, color=&quot;Koala&quot;, 
                                    axes = c(1, 3)) 
unwt.unifrac.1.3.ml &lt;- plot_ordination(ps.rar.tablefilter.ml, 
                                    ord.unw.uni.ml, color=&quot;Koala&quot;, 
                                    axes = c(1, 3)) 

w.unifrac.1.3.cl &lt;- plot_ordination(ps.rar.tablefilter.cl, 
                                 ord.w.uni.cl, color=&quot;Koala&quot;, 
                                 axes = c(1, 3))
w.unifrac.1.3.ml &lt;- plot_ordination(ps.rar.tablefilter.ml, 
                                 ord.w.uni.ml, color=&quot;Koala&quot;, 
                                 axes = c(1, 3))


#Create theme
theme_pcoa &lt;- theme(axis.text.x = element_text(face=&quot;bold&quot;, size=16), 
                    axis.text.y = element_text(face=&quot;bold&quot;, size=16),
                    axis.title.x = element_text(size=20, face=&quot;bold&quot;),
                    axis.title.y = element_text(size=20, face=&quot;bold&quot;),
                    axis.line = element_line(colour = &quot;black&quot;),
                    #Background panel
                    panel.background = element_rect(fill = &quot;White&quot;),
                    panel.grid.major = element_line(colour = &quot;white&quot;), 
                    panel.grid.minor = element_line(colour = &quot;white&quot;),
                    #Legend
                    legend.title = element_blank(),
                    legend.text = element_text(size=16),
                    legend.key = element_rect(fill = &quot;white&quot;, color = NA),
                    legend.key.size = unit(2.5, &quot;line&quot;))
          

#Run through plots
#Cleland only

cbp2_beta &lt;- c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;, 
               &quot;#F0E442&quot;, &quot;#0072B2&quot;, &quot;#D55E00&quot;)
cl.uw.1.2 &lt;- unwt.unifrac.1.2.cl +
  scale_x_reverse() +
  geom_point(size=7.5) +
  geom_path() +
  geom_text(aes(label = Collection_number), color = &quot;black&quot;, fontface = &quot;bold&quot;) +
  scale_colour_manual(values=cbp2_beta) +
  theme_pcoa +
  theme(legend.position = &quot;none&quot;)

cl.w.1.2 &lt;- w.unifrac.1.2.cl +
  scale_x_reverse() +
  geom_point(size=7.5) +
  geom_path() +
  geom_text(aes(label = Collection_number), color = &quot;black&quot;, fontface = &quot;bold&quot;) +
  scale_colour_manual(values=cbp2_beta) +
  theme_pcoa +
  theme(legend.position = &quot;right&quot;)


#MtLagoon only
cbp2_beta &lt;- c(&quot;#88CCEE&quot;, &quot;#CC6677&quot;, &quot;#DDCC77&quot;, &quot;#117733&quot;, &quot;#998ec3&quot;)

ml.uw.1.2 &lt;- unwt.unifrac.1.2.ml +
  scale_x_reverse() +
  geom_point(size=7.5) +
  geom_path() +
  geom_text(aes(label = Collection_number), color = &quot;black&quot;, fontface = &quot;bold&quot;) +
  scale_colour_manual(values=cbp2_beta) +
  theme_pcoa +
  theme(legend.position = &quot;none&quot;)

ml.w.1.2 &lt;- w.unifrac.1.2.ml +
  scale_x_reverse() +
  geom_point(size=7.5) +
  geom_path() +
  geom_text(aes(label = Collection_number), color = &quot;black&quot;, fontface = &quot;bold&quot;) +
  scale_colour_manual(values=cbp2_beta) +
  theme_pcoa +
  theme(legend.position = &quot;right&quot;)

#Join into SI figure X
ggarrange(cl.uw.1.2, cl.w.1.2, ml.uw.1.2, ml.w.1.2,
          nrow = 2, ncol = 2, common.legend = FALSE, labels = c(&quot;A)&quot;, &quot;B)&quot;, &quot;C)&quot;, &quot;D)&quot;), 
          font.label = list(size=18, face=&quot;bold&quot;, color=&quot;black&quot;))</code></pre>
<p><img src="figure/Supplemental_analysis_for_reviewer1.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(filename = &quot;output/Figure_3.pdf&quot;, 
#        width = 16, height = 10, dpi = 300)</code></pre>
</div>
<div id="beta-diversity-stats" class="section level1">
<h1>Beta-diversity stats</h1>
<pre class="r"><code>library(vegan)</code></pre>
<pre><code>Loading required package: permute</code></pre>
<pre><code>Loading required package: lattice</code></pre>
<pre><code>This is vegan 2.5-7</code></pre>
<pre><code>
Attaching package: &#39;vegan&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:microbiome&#39;:

    diversity</code></pre>
<pre class="r"><code>metadata.cl &lt;- as(sample_data(ps.rar.tablefilter.cl), &quot;data.frame&quot;)
metadata.ml &lt;- as(sample_data(ps.rar.tablefilter.ml), &quot;data.frame&quot;)

adonis(distance(ps.rar.tablefilter.cl, method=&quot;unifrac&quot;) ~ Koala, data = metadata.cl)</code></pre>
<pre><code>
Call:
adonis(formula = distance(ps.rar.tablefilter.cl, method = &quot;unifrac&quot;) ~      Koala, data = metadata.cl) 

Permutation: free
Number of permutations: 999

Terms added sequentially (first to last)

          Df SumsOfSqs MeanSqs F.Model      R2 Pr(&gt;F)    
Koala      6    2.9229 0.48715  28.245 0.77571  0.001 ***
Residuals 49    0.8451 0.01725         0.22429           
Total     55    3.7680                 1.00000           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>adonis(distance(ps.rar.tablefilter.ml, method=&quot;unifrac&quot;) ~ Koala, data = metadata.ml)</code></pre>
<pre><code>
Call:
adonis(formula = distance(ps.rar.tablefilter.ml, method = &quot;unifrac&quot;) ~      Koala, data = metadata.ml) 

Permutation: free
Number of permutations: 999

Terms added sequentially (first to last)

          Df SumsOfSqs  MeanSqs F.Model      R2 Pr(&gt;F)    
Koala      4   0.76829 0.192073  13.514 0.73992  0.001 ***
Residuals 19   0.27005 0.014213         0.26008           
Total     23   1.03834                  1.00000           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>adonis(distance(ps.rar.tablefilter.cl, method=&quot;wunifrac&quot;) ~ Koala, data = metadata.cl)</code></pre>
<pre><code>
Call:
adonis(formula = distance(ps.rar.tablefilter.cl, method = &quot;wunifrac&quot;) ~      Koala, data = metadata.cl) 

Permutation: free
Number of permutations: 999

Terms added sequentially (first to last)

          Df SumsOfSqs MeanSqs F.Model      R2 Pr(&gt;F)    
Koala      6    1.0544 0.17573  24.139 0.74721  0.001 ***
Residuals 49    0.3567 0.00728         0.25279           
Total     55    1.4111                 1.00000           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>adonis(distance(ps.rar.tablefilter.ml, method=&quot;wunifrac&quot;) ~ Koala, data = metadata.ml)</code></pre>
<pre><code>
Call:
adonis(formula = distance(ps.rar.tablefilter.ml, method = &quot;wunifrac&quot;) ~      Koala, data = metadata.ml) 

Permutation: free
Number of permutations: 999

Terms added sequentially (first to last)

          Df SumsOfSqs  MeanSqs F.Model      R2 Pr(&gt;F)    
Koala      4  0.194845 0.048711  14.811 0.75717  0.001 ***
Residuals 19  0.062487 0.003289         0.24283           
Total     23  0.257331                  1.00000           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<div id="taxonomic-bar-plots-figure-4" class="section level1">
<h1>Taxonomic bar plots (Figure 4)</h1>
<pre class="r"><code>library(ggh4x)
library(ggtext)

#Collapse to family level
ps.family &lt;- tax_glom(ps.rar.tablefilter, taxrank = &quot;Family&quot;, NArm = FALSE)

#Extract top 20 most abundant family names, bind to ps sampledata
top20families = names(sort(taxa_sums(ps.family), TRUE)[1:20])
taxtab20 = cbind(tax_table(ps.family), family_20 = NA)
taxtab20[top20families, &quot;family_20&quot;] &lt;- as(tax_table(ps.family)
                                           [top20families, &quot;Family&quot;], &quot;character&quot;)
tax_table(ps.family) &lt;- tax_table(taxtab20)

ps.family.ra &lt;- transform_sample_counts(ps.family, function(x) 100 * x/sum(x))

#Melt into a dataframe
pd.family &lt;- psmelt(ps.family.ra)

#Replace NA with &#39;other&#39;, for plotting purposes
pd.family &lt;- arrange(pd.family, Koala_Collection)
pd.family$family_20[is.na(pd.family$family_20)] &lt;- c(&quot;Other&quot;)

#Relative abundance of top 20 families?
mean(sample_sums(prune_taxa(top20families, ps.family.ra)))</code></pre>
<pre><code>[1] 98.51737</code></pre>
<pre class="r"><code>#Plot em
#pd.family.plot &lt;- 
  
  ggplot(pd.family, aes(x = Koala_Collection, y = Abundance,
                                        fill = fct_reorder(family_20, -Abundance))) +
  geom_bar(width = 0.9, stat = &quot;identity&quot;) +
  facet_nested(~Location_sampled + Koala, scales = &quot;free&quot;, space = &quot;free&quot;, ) +
  labs(x = &quot;&quot;, y = &quot;Relative abundance&quot;) +
  theme(
        axis.text.y = element_text(size=16, face = &#39;bold&#39;),
        axis.title.y = element_text(size=16, face = &#39;bold&#39;),
        axis.ticks.y = element_line(size = 1),
        axis.ticks.x = element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        legend.title = element_blank(),
        legend.text = element_text(size = 18),
        legend.position = &quot;bottom&quot;,
        strip.background = element_blank(),
        strip.text = element_textbox_simple(
          padding = margin(5, 0, 5, 0),
          margin = margin(5, 5, 5, 5),
          size = 14,
          face = &quot;bold&quot;,
          halign = 0.5,
          fill = &quot;white&quot;,
          box.color = &quot;grey&quot;,
          linewidth = 1.5,
          linetype = &quot;solid&quot;,),
        panel.background = element_blank()
        ) +
  scale_fill_manual(values = 
      c(&quot;Bacteroidaceae&quot; = &quot;#228B22&quot;, &quot;[Eubacterium]_coprostanoligenes_group&quot; = &quot;#3CB371&quot;,
        &quot;Akkermansiaceae&quot; = &quot;#2E8B57&quot;, &quot;Tannerellaceae&quot; = &quot;#8FBC8F&quot;,
        &quot;Christensenellaceae&quot; = &quot;#191970&quot;, &quot;Clostridiaceae&quot; = &quot;#db6d00&quot;,
        &quot;Paludibacteraceae&quot; = &quot;#66CCEE&quot;, &quot;Muribaculaceae&quot; = &quot;#FF1493&quot;,
        &quot;Erysipelatoclostridiaceae&quot; = &quot;#0000FF&quot;, &quot;Lachnospiraceae&quot; = &quot;#B8860B&quot;,
        &quot;Oscillospiraceae&quot; = &quot;#FF6347&quot;, &quot;Acidaminococcaceae&quot; = &quot;#F0E68C&quot;,
        &quot;Pasteurellaceae&quot; = &quot;#FF69B4&quot;, &quot;Desulfovibrionaceae&quot; = &quot;#FFC0CB&quot;,
        &quot;Prevotellaceae&quot; = &quot;#FFFF00&quot;, &quot;Rikenellaceae&quot; = &quot;#00FF7F&quot;,
        &quot;Ruminococcaceae&quot; = &quot;#b66dff&quot;, &quot;vadinHA49&quot; = &quot;#FF00FF&quot;,
        &quot;Synergistaceae&quot; = &quot;#8B0000&quot;, &quot;Gastranaerophilales&quot; = &quot;#000000&quot;, &quot;Other&quot; = &quot;#808080&quot;)
  )</code></pre>
<p><img src="figure/Supplemental_analysis_for_reviewer1.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(filename = &quot;output/Figure_4.pdf&quot;, 
#        width = 20, height = 12, dpi = 300)</code></pre>
</div>
<div
id="mean-relative-abundances-of-select-microbial-families-between-clml-main-text"
class="section level1">
<h1>Mean relative abundances of select microbial families between CL/ML
(main text)</h1>
<pre class="r"><code>#Merge samples by geographic location
ps.family.merged &lt;- merge_samples(ps.family, &quot;Location_sampled&quot;)</code></pre>
<pre><code>Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion

Warning in asMethod(object): NAs introduced by coercion</code></pre>
<pre class="r"><code>ps.family.merged.relab &lt;- transform_sample_counts(ps.family.merged, function(x) x / sum(x))
ps.family.merged.relab@otu_table &lt;- phyloseq::t(ps.family.merged.relab@otu_table)
#Had to transpose otu table, as merge_samples flips it for some reason...

ps.family.merged.relab.tax &lt;- as.data.frame(tax_table(ps.family.merged.relab))
ps.family.merged.relab.otu.and.tax &lt;- cbind(ps.family.merged.relab.tax, 
                                              ps.family.merged.relab@otu_table)

DT::datatable(ps.family.merged.relab.otu.and.tax)</code></pre>
<div id="htmlwidget-bfbac5dbd66252231703" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-bfbac5dbd66252231703">{"x":{"filter":"none","vertical":false,"data":[["b774fb602227bcb52e612a5a44edebdb","b825c0a245773b808a0c9f4ac793eb8d","33c12fd798ba06fef726af2064fa8349","9807b2b33295a8a1f2780f6138b09ce7","e28e192de195e51ce8d263e61bcd4a06","640c940d20a5ae191c28fc60628852f7","e314819c1db74a311d8d31f593130005","e6284bf8aaac91ea2648771704bc874e","ce3b37d15aee37e4213c2d13ad8b1b1a","b2e212df9eae75b283a9ac072bcb1887","b445b7b50d9a81353b3a06f5a3546118","2055090bb53979c18a524e2e07b82b36","c7114b8079ec44f57ea317d2cf019364","593a5c45dd38f6d9ec369b9b2679997f","aa06565a40c388728c071e1aa9a75ff3","1558e3c18758f8706ea6214b3b330c5c","e2a44f0a3bc38ae866581674338236af","248b8cab587a261ad4b9e60ca6aa7c18","75282d0a08867be324a5ad58ba522b95","7a4a637b2896871956bbfb99b299beec","3afcfc5fdf190d03a44434bd5a069f7f","91e878b3df6781270d5567086ad250c0","43458b741d6ea747d2cc9c6259505bff","3045e0403c60e0014c1550f961a7e173","8827b464dcb48eba9a045d2a9d193d48","ffa75974fef49864e3aa893cb9034b16","ffd03dbe2d1bbb900f79153a0944a058","3f90b85405c28eba22ec578de3dc45c9","6f369d59b822656097960c7c6550facc","0d426c1c43d3948c6c3cd545c716055c","724fe77ff54fcda16bf32851a71114b2","4d38774a175ac1bc9e7f28e9279e2883"],["d__Archaea","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria","d__Bacteria"],["Euryarchaeota","Proteobacteria","Cyanobacteria","Proteobacteria","Desulfobacterota","Rs-K70_termite_group","Fusobacteriota","Firmicutes","Bacteroidota","Bacteroidota","Firmicutes","Bacteroidota","Bacteroidota","Bacteroidota",null,"Bacteroidota","Synergistota","Verrucomicrobiota","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes","Planctomycetota","Bacteroidota","Proteobacteria","Proteobacteria","Proteobacteria"],["Methanobacteria","Gammaproteobacteria","Vampirivibrionia","Alphaproteobacteria","Desulfovibrionia","Rs-K70_termite_group","Fusobacteriia","Bacilli","Bacteroidia","Bacteroidia","Clostridia","Bacteroidia","Bacteroidia","Bacteroidia",null,"Bacteroidia","Synergistia","Verrucomicrobiae","Clostridia","Clostridia","Clostridia","Negativicutes","Negativicutes","Negativicutes","Clostridia","Clostridia","Clostridia","vadinHA49","Bacteroidia","Gammaproteobacteria","Gammaproteobacteria","Gammaproteobacteria"],["Methanobacteriales","Burkholderiales","Gastranaerophilales","Rhodospirillales","Desulfovibrionales","Rs-K70_termite_group","Fusobacteriales","Erysipelotrichales","Bacteroidales","Bacteroidales","Lachnospirales","Bacteroidales","Bacteroidales","Bacteroidales",null,"Bacteroidales","Synergistales","Verrucomicrobiales","Oscillospirales","Oscillospirales","Clostridia_vadinBB60_group","Acidaminococcales","Veillonellales-Selenomonadales","Veillonellales-Selenomonadales","Oscillospirales","Oscillospirales","Monoglobales","vadinHA49","Bacteroidales","Burkholderiales","Pasteurellales","Enterobacterales"],["Methanobacteriaceae","Oxalobacteraceae","Gastranaerophilales","uncultured","Desulfovibrionaceae","Rs-K70_termite_group","Fusobacteriaceae","Erysipelatoclostridiaceae","Paludibacteraceae","Marinifilaceae","Lachnospiraceae","Prevotellaceae","Muribaculaceae","Tannerellaceae",null,"Bacteroidaceae","Synergistaceae","Akkermansiaceae","[Eubacterium]_coprostanoligenes_group","Ruminococcaceae","Clostridia_vadinBB60_group","Acidaminococcaceae","Selenomonadaceae","Veillonellaceae","Butyricicoccaceae","Oscillospiraceae","Monoglobaceae","vadinHA49","Rikenellaceae","Sutterellaceae","Pasteurellaceae","Enterobacteriaceae"],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[null,null,"Gastranaerophilales",null,"Desulfovibrionaceae",null,"Fusobacteriaceae","Erysipelatoclostridiaceae","Paludibacteraceae",null,"Lachnospiraceae","Prevotellaceae","Muribaculaceae","Tannerellaceae",null,"Bacteroidaceae","Synergistaceae","Akkermansiaceae","[Eubacterium]_coprostanoligenes_group","Ruminococcaceae","Clostridia_vadinBB60_group","Acidaminococcaceae",null,null,null,"Oscillospiraceae",null,null,"Rikenellaceae",null,"Pasteurellaceae",null],[0,4.65875141497184e-05,0.0672798046099832,0.00123704718423082,0.0114709363297365,0,0.00483122433969633,0.00527231888856068,0,0,0.127255281735514,0.0107835226901018,0.126282891280602,0.265414519841325,0.00881594363505522,0.0600086831196585,0.00984285139375965,0.0153847831301655,0.00662087873963923,0.126944037492045,0.00491894763761654,0.00135748086442637,0.000559545781649277,0.000294889052330664,0.00032165209237412,0.114745542962609,0.00210734159749577,0.000323634539784746,0.0112008278700387,0.00312780640211567,0.0109500482725944,0.00260097100274172],[0.00133974372335276,0.00187008307105852,0.000926356939224037,0,0.00403660036266874,0.00487842723118859,0,0.00154354224998205,0.158325980375128,0.000569709517622783,0.0667451754172659,0,0,0.378461969573806,0.000633396557194435,0.00640460028856019,0.111507900666745,0,0.00189555788688719,0.0289521281892733,0.00108615351124018,0.0476112728375934,0,0.0053068673155797,0,0.0637264097415696,0,0.00987264907978018,0.0898855717590823,0,0.0136938714540793,0.000726032251116839]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Kingdom<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n      <th>Genus<\/th>\n      <th>Species<\/th>\n      <th>family_20<\/th>\n      <th>Cleland<\/th>\n      <th>Mountain_Lagoon<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="venn-diagram-si-figure-x" class="section level1">
<h1>Venn diagram (SI figure X)</h1>
<pre class="r"><code>ps.rar.tablefilter.cl.table &lt;- otu_table(ps.rar.tablefilter.cl)
ps.rar.tablefilter.ml.table &lt;- otu_table(ps.rar.tablefilter.ml)

#For each ASV (row), if abundance &gt; 0, print ASV (rowname) to a vector 
venn.CL &lt;- rownames(ps.rar.tablefilter.cl.table[ apply(ps.rar.tablefilter.cl.table, MARGIN = 1, function(x) any(x &gt; 0))])

venn.ML &lt;- rownames(ps.rar.tablefilter.ml.table[ apply(ps.rar.tablefilter.ml.table, MARGIN = 1, function(x) any(x &gt; 0))])

ppn.comparison &lt;- list(&quot;Cleland&quot; = venn.CL, &quot;Mountain Lagoon&quot; = venn.ML)

venn.diagram(ppn.comparison, print.mode = c(&quot;raw&quot;,&quot;percent&quot;), fill = c(&quot;red&quot;, &quot;blue&quot;), inverted = TRUE, 
             imagetype = &quot;png&quot;, filename = &quot;output/SI_figure3.png&quot;, cat.pos = c(0,0))</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>#What are the dominant ASVs that are shared between ppns?
shared.asvs &lt;- inner_join(as.data.frame(venn.CL), as.data.frame(venn.ML), by = c(&quot;venn.CL&quot; = &quot;venn.ML&quot;))

shared.asvs.vector &lt;- as.character(shared.asvs$venn.CL)

#Mean relative abundance of shared ASVs
percent(mean(sample_sums(prune_taxa(shared.asvs.vector, ps.rar.tablefilter)))/
  mean(sample_sums(ps.rar.tablefilter)))</code></pre>
<pre><code>[1] &quot;33%&quot;</code></pre>
</div>
<div id="asv-level-analyses-table-2" class="section level1">
<h1>ASV-level analyses (Table 2)</h1>
<pre class="r"><code>library(gridExtra)</code></pre>
<pre><code>
Attaching package: &#39;gridExtra&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    combine</code></pre>
<pre class="r"><code>options(digits=3)

##How many ASVs are always present in a given koala through time?
#Create a list of each subset that I want (per koala)
koala_list &lt;- c(
  ps.blaze &lt;- subset_samples(ps.rar.tablefilter, Koala==&quot;Blaze&quot;),
  ps.byron &lt;- subset_samples(ps.rar.tablefilter, Koala==&quot;Byron&quot;),
  ps.cin &lt;- subset_samples(ps.rar.tablefilter, Koala==&quot;Cin&quot;),
  ps.claudette &lt;- subset_samples(ps.rar.tablefilter, Koala==&quot;Claudette&quot;),
  ps.george &lt;- subset_samples(ps.rar.tablefilter, Koala==&quot;George&quot;),
  ps.leo &lt;- subset_samples(ps.rar.tablefilter, Koala==&quot;Leo&quot;),
  ps.lola &lt;- subset_samples(ps.rar.tablefilter, Koala==&quot;Lola&quot;),
  ps.phasco &lt;- subset_samples(ps.rar.tablefilter, Koala==&quot;Phasco&quot;),
  ps.reus &lt;- subset_samples(ps.rar.tablefilter, Koala==&quot;Reus&quot;),
  ps.theo &lt;- subset_samples(ps.rar.tablefilter, Koala==&quot;Theo&quot;),
  ps.tos &lt;- subset_samples(ps.rar.tablefilter, Koala==&quot;Tos&quot;),
  ps.uno &lt;- subset_samples(ps.rar.tablefilter, Koala==&quot;Uno&quot;)
)

#Create functions to use in loop
gen_x &lt;- function(ps){
  print(length(taxa_names(filter_taxa(ps, function(x) sum(x) &gt; 0 , TRUE))))
}
gen_y &lt;- function(ps){
  print(length(taxa_names(core(ps, detection =1, prevalence =0.999))))
}
gen_z &lt;- function(ps){
  print(mean(sample_sums(core(ps, detection =1, prevalence =0.999))) / mean(sample_sums(ps)))
}
gen_zz &lt;- function(ps){
  print(length(taxa_names(core(ps, detection = 1, prevalence = 0))) - length(taxa_names(core(ps, detection = 1, prevalence = 1/length(sample_sums(ps))))))
}





#Create dataframe of length koalalist
ASVresults &lt;- data.frame(totalASVs_per_koala=rep(0,length(koala_list)),
                      total_core_ASVs=rep(0,length(koala_list)),
                      mean_rel_abun_core_ASVs=rep(0,length(koala_list)),
                      num_singleton_ASVs_per_koala=rep(0, length(koala_list)))
row.names(ASVresults) &lt;- sort(unique(ps.rar@sam_data$Koala))

#Loop over each subsetted phyloseq object, outputting results into the dataframe!
for (ps in 1:length(koala_list)){
  x &lt;- gen_x(koala_list[[ps]])
  y &lt;- gen_y(koala_list[[ps]])
  z &lt;- gen_z(koala_list[[ps]])
  zz &lt;- gen_zz(koala_list[[ps]])
  ASVresults[ps, 1] &lt;- x
  ASVresults[ps, 2] &lt;- y
  ASVresults[ps, 3] &lt;- z
  ASVresults[ps, 4] &lt;- zz
}</code></pre>
<pre><code>[1] 98
[1] 42
[1] 0.993
[1] 16
[1] 120
[1] 53
[1] 0.916
[1] 17
[1] 58
[1] 33
[1] 0.976
[1] 9
[1] 101
[1] 44
[1] 0.992
[1] 17
[1] 120
[1] 65
[1] 0.973
[1] 18
[1] 83
[1] 45
[1] 0.995
[1] 10
[1] 127
[1] 48
[1] 0.992
[1] 42
[1] 58
[1] 45
[1] 0.994
[1] 0
[1] 50
[1] 40
[1] 0.964
[1] 5
[1] 111
[1] 51
[1] 0.942
[1] 13
[1] 58
[1] 31
[1] 0.993
[1] 6
[1] 61
[1] 29
[1] 0.987
[1] 9</code></pre>
<pre class="r"><code>knitr::kable(
  ASVresults
)</code></pre>
<table>
<colgroup>
<col width="10%" />
<col width="20%" />
<col width="16%" />
<col width="24%" />
<col width="29%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">totalASVs_per_koala</th>
<th align="right">total_core_ASVs</th>
<th align="right">mean_rel_abun_core_ASVs</th>
<th align="right">num_singleton_ASVs_per_koala</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Blaze</td>
<td align="right">98</td>
<td align="right">42</td>
<td align="right">0.993</td>
<td align="right">16</td>
</tr>
<tr class="even">
<td align="left">Byron</td>
<td align="right">120</td>
<td align="right">53</td>
<td align="right">0.916</td>
<td align="right">17</td>
</tr>
<tr class="odd">
<td align="left">Cin</td>
<td align="right">58</td>
<td align="right">33</td>
<td align="right">0.976</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">Claudette</td>
<td align="right">101</td>
<td align="right">44</td>
<td align="right">0.992</td>
<td align="right">17</td>
</tr>
<tr class="odd">
<td align="left">George</td>
<td align="right">120</td>
<td align="right">65</td>
<td align="right">0.973</td>
<td align="right">18</td>
</tr>
<tr class="even">
<td align="left">Leo</td>
<td align="right">83</td>
<td align="right">45</td>
<td align="right">0.995</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">Lola</td>
<td align="right">127</td>
<td align="right">48</td>
<td align="right">0.992</td>
<td align="right">42</td>
</tr>
<tr class="even">
<td align="left">Phasco</td>
<td align="right">58</td>
<td align="right">45</td>
<td align="right">0.994</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Reus</td>
<td align="right">50</td>
<td align="right">40</td>
<td align="right">0.964</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">Theo</td>
<td align="right">111</td>
<td align="right">51</td>
<td align="right">0.942</td>
<td align="right">13</td>
</tr>
<tr class="odd">
<td align="left">Tos</td>
<td align="right">58</td>
<td align="right">31</td>
<td align="right">0.993</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">Uno</td>
<td align="right">61</td>
<td align="right">29</td>
<td align="right">0.987</td>
<td align="right">9</td>
</tr>
</tbody>
</table>
</div>
<div id="lonepinella-prevalanceabundance" class="section level1">
<h1>Lonepinella prevalance/abundance</h1>
<pre class="r"><code>#Get ASV IDs
lp_ids &lt;- tax_table(ps.rar.tablefilter) %&gt;%
                as.data.frame() %&gt;%
                filter(Genus == &quot;Lonepinella&quot;) %&gt;%
                row.names()

#Calculate prevalence, min, max, and mean relative abundance
ps.lp.gen &lt;- prune_taxa(lp_ids, ps.rar.tablefilter) %&gt;%
  tax_glom(taxrank = &quot;Genus&quot;, NArm = FALSE)</code></pre>
<pre><code>Warning in FUN(X[[i]], ...): merge_taxa attempted to reduce tree to 1 or fewer tips.
 tree replaced with NULL.</code></pre>
<pre class="r"><code>ps.rar.gen &lt;- tax_glom(ps.rar.tablefilter, taxrank = &quot;Genus&quot;, NArm = FALSE)

mean(sample_sums(ps.lp.gen)/sample_sums(ps.rar.gen)) %&gt;%
  percent(accuracy = 0.01)</code></pre>
<pre><code>[1] &quot;1.17%&quot;</code></pre>
<pre class="r"><code>min(sample_sums(ps.lp.gen)/sample_sums(ps.rar.gen)) %&gt;%
  percent(accuracy = 0.01)</code></pre>
<pre><code>[1] &quot;0.00%&quot;</code></pre>
<pre class="r"><code>max(sample_sums(ps.lp.gen)/sample_sums(ps.rar.gen)) %&gt;%
  percent(accuracy = 0.01)</code></pre>
<pre><code>[1] &quot;9.77%&quot;</code></pre>
<pre class="r"><code>prevalence(ps.lp.gen)</code></pre>
<pre><code>724fe77ff54fcda16bf32851a71114b2 
                           0.825 </code></pre>
<p>#Software/package versions</p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
[3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.1252    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gridExtra_2.3       ggtext_0.1.1        ggh4x_0.2.1        
 [4] vegan_2.5-7         lattice_0.20-45     permute_0.9-5      
 [7] stargazer_5.2.2     lmerTest_3.1-3      lme4_1.1-27.1      
[10] Matrix_1.3-4        osmdata_0.1.8       VennDiagram_1.7.1  
[13] futile.logger_1.4.3 ggpubr_0.4.0        ggrepel_0.9.1      
[16] ggmap_3.0.0         forcats_0.5.1       stringr_1.4.0      
[19] dplyr_1.0.7         purrr_0.3.4         readr_2.1.1        
[22] tidyr_1.1.4         tibble_3.1.6        tidyverse_1.3.1    
[25] microbiome_1.16.0   scales_1.1.1        ggplot2_3.3.5      
[28] qiime2R_0.99.6      phyloseq_1.38.0    

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.4.1        Hmisc_4.6-0           
  [4] workflowr_1.7.0        plyr_1.8.6             igraph_1.2.11         
  [7] sp_1.4-6               splines_4.1.2          crosstalk_1.2.0       
 [10] GenomeInfoDb_1.30.0    digest_0.6.29          foreach_1.5.1         
 [13] htmltools_0.5.2        fansi_0.5.0            magrittr_2.0.1        
 [16] checkmate_2.0.0        cluster_2.1.2          tzdb_0.2.0            
 [19] Biostrings_2.62.0      modelr_0.1.8           jpeg_0.1-9            
 [22] colorspace_2.0-2       rvest_1.0.2            haven_2.4.3           
 [25] xfun_0.29              crayon_1.4.2           RCurl_1.98-1.5        
 [28] jsonlite_1.7.2         survival_3.2-13        iterators_1.0.13      
 [31] ape_5.6-1              glue_1.6.0             gtable_0.3.0          
 [34] zlibbioc_1.40.0        XVector_0.34.0         car_3.0-12            
 [37] Rhdf5lib_1.16.0        BiocGenerics_0.40.0    abind_1.4-5           
 [40] futile.options_1.0.1   DBI_1.1.2              rstatix_0.7.0         
 [43] Rcpp_1.0.7             gridtext_0.1.4         htmlTable_2.4.0       
 [46] foreign_0.8-81         Formula_1.2-4          stats4_4.1.2          
 [49] DT_0.20                truncnorm_1.0-8        htmlwidgets_1.5.4     
 [52] httr_1.4.2             RColorBrewer_1.1-2     ellipsis_0.3.2        
 [55] farver_2.1.0           pkgconfig_2.0.3        NADA_1.6-1.1          
 [58] nnet_7.3-16            sass_0.4.2             dbplyr_2.1.1          
 [61] utf8_1.2.2             labeling_0.4.2         tidyselect_1.1.1      
 [64] rlang_0.4.12           reshape2_1.4.4         later_1.3.0           
 [67] munsell_0.5.0          cellranger_1.1.0       tools_4.1.2           
 [70] cachem_1.0.6           cli_3.1.0              generics_0.1.1        
 [73] ade4_1.7-18            broom_0.7.11           evaluate_0.14         
 [76] biomformat_1.22.0      fastmap_1.1.0          yaml_2.2.1            
 [79] knitr_1.37             fs_1.5.2               RgoogleMaps_1.4.5.3   
 [82] nlme_3.1-153           formatR_1.11           xml2_1.3.3            
 [85] compiler_4.1.2         rstudioapi_0.13        curl_4.3.2            
 [88] png_0.1-7              ggsignif_0.6.3         zCompositions_1.3.4   
 [91] reprex_2.0.1           bslib_0.4.0            stringi_1.7.6         
 [94] highr_0.9              markdown_1.1           nloptr_1.2.2.3        
 [97] multtest_2.50.0        vctrs_0.3.8            pillar_1.6.4          
[100] lifecycle_1.0.1        rhdf5filters_1.6.0     jquerylib_0.1.4       
[103] cowplot_1.1.1          data.table_1.14.2      bitops_1.0-7          
[106] httpuv_1.6.5           R6_2.5.1               latticeExtra_0.6-29   
[109] promises_1.2.0.1       IRanges_2.28.0         codetools_0.2-18      
[112] lambda.r_1.2.4         boot_1.3-28            MASS_7.3-54           
[115] assertthat_0.2.1       rhdf5_2.38.0           rprojroot_2.0.2       
[118] rjson_0.2.21           withr_2.4.3            S4Vectors_0.32.3      
[121] GenomeInfoDbData_1.2.7 mgcv_1.8-38            parallel_4.1.2        
[124] hms_1.1.1              rpart_4.1-15           minqa_1.2.4           
[127] rmarkdown_2.11         carData_3.0-5          Rtsne_0.15            
[130] git2r_0.29.0           numDeriv_2016.8-1.1    Biobase_2.54.0        
[133] lubridate_1.8.0        base64enc_0.1-3       </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
[3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.1252    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gridExtra_2.3       ggtext_0.1.1        ggh4x_0.2.1        
 [4] vegan_2.5-7         lattice_0.20-45     permute_0.9-5      
 [7] stargazer_5.2.2     lmerTest_3.1-3      lme4_1.1-27.1      
[10] Matrix_1.3-4        osmdata_0.1.8       VennDiagram_1.7.1  
[13] futile.logger_1.4.3 ggpubr_0.4.0        ggrepel_0.9.1      
[16] ggmap_3.0.0         forcats_0.5.1       stringr_1.4.0      
[19] dplyr_1.0.7         purrr_0.3.4         readr_2.1.1        
[22] tidyr_1.1.4         tibble_3.1.6        tidyverse_1.3.1    
[25] microbiome_1.16.0   scales_1.1.1        ggplot2_3.3.5      
[28] qiime2R_0.99.6      phyloseq_1.38.0    

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.4.1        Hmisc_4.6-0           
  [4] workflowr_1.7.0        plyr_1.8.6             igraph_1.2.11         
  [7] sp_1.4-6               splines_4.1.2          crosstalk_1.2.0       
 [10] GenomeInfoDb_1.30.0    digest_0.6.29          foreach_1.5.1         
 [13] htmltools_0.5.2        fansi_0.5.0            magrittr_2.0.1        
 [16] checkmate_2.0.0        cluster_2.1.2          tzdb_0.2.0            
 [19] Biostrings_2.62.0      modelr_0.1.8           jpeg_0.1-9            
 [22] colorspace_2.0-2       rvest_1.0.2            haven_2.4.3           
 [25] xfun_0.29              crayon_1.4.2           RCurl_1.98-1.5        
 [28] jsonlite_1.7.2         survival_3.2-13        iterators_1.0.13      
 [31] ape_5.6-1              glue_1.6.0             gtable_0.3.0          
 [34] zlibbioc_1.40.0        XVector_0.34.0         car_3.0-12            
 [37] Rhdf5lib_1.16.0        BiocGenerics_0.40.0    abind_1.4-5           
 [40] futile.options_1.0.1   DBI_1.1.2              rstatix_0.7.0         
 [43] Rcpp_1.0.7             gridtext_0.1.4         htmlTable_2.4.0       
 [46] foreign_0.8-81         Formula_1.2-4          stats4_4.1.2          
 [49] DT_0.20                truncnorm_1.0-8        htmlwidgets_1.5.4     
 [52] httr_1.4.2             RColorBrewer_1.1-2     ellipsis_0.3.2        
 [55] farver_2.1.0           pkgconfig_2.0.3        NADA_1.6-1.1          
 [58] nnet_7.3-16            sass_0.4.2             dbplyr_2.1.1          
 [61] utf8_1.2.2             labeling_0.4.2         tidyselect_1.1.1      
 [64] rlang_0.4.12           reshape2_1.4.4         later_1.3.0           
 [67] munsell_0.5.0          cellranger_1.1.0       tools_4.1.2           
 [70] cachem_1.0.6           cli_3.1.0              generics_0.1.1        
 [73] ade4_1.7-18            broom_0.7.11           evaluate_0.14         
 [76] biomformat_1.22.0      fastmap_1.1.0          yaml_2.2.1            
 [79] knitr_1.37             fs_1.5.2               RgoogleMaps_1.4.5.3   
 [82] nlme_3.1-153           formatR_1.11           xml2_1.3.3            
 [85] compiler_4.1.2         rstudioapi_0.13        curl_4.3.2            
 [88] png_0.1-7              ggsignif_0.6.3         zCompositions_1.3.4   
 [91] reprex_2.0.1           bslib_0.4.0            stringi_1.7.6         
 [94] highr_0.9              markdown_1.1           nloptr_1.2.2.3        
 [97] multtest_2.50.0        vctrs_0.3.8            pillar_1.6.4          
[100] lifecycle_1.0.1        rhdf5filters_1.6.0     jquerylib_0.1.4       
[103] cowplot_1.1.1          data.table_1.14.2      bitops_1.0-7          
[106] httpuv_1.6.5           R6_2.5.1               latticeExtra_0.6-29   
[109] promises_1.2.0.1       IRanges_2.28.0         codetools_0.2-18      
[112] lambda.r_1.2.4         boot_1.3-28            MASS_7.3-54           
[115] assertthat_0.2.1       rhdf5_2.38.0           rprojroot_2.0.2       
[118] rjson_0.2.21           withr_2.4.3            S4Vectors_0.32.3      
[121] GenomeInfoDbData_1.2.7 mgcv_1.8-38            parallel_4.1.2        
[124] hms_1.1.1              rpart_4.1-15           minqa_1.2.4           
[127] rmarkdown_2.11         carData_3.0-5          Rtsne_0.15            
[130] git2r_0.29.0           numDeriv_2016.8-1.1    Biobase_2.54.0        
[133] lubridate_1.8.0        base64enc_0.1-3       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
